lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -816,6 +816,22 @@ ::mlir::LogicalResult mlir::tt::ttir::TransposeOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// TypecastOp
+//===----------------------------------------------------------------------===//
+
+// TypecastOp folder
+::llvm::LogicalResult mlir::tt::ttir::TypecastOp::fold(
+    FoldAdaptor adaptor,
+    ::llvm::SmallVectorImpl<::mlir::OpFoldResult> &results) {
+
+  if (getType(0) == getInputs()[0].getType()) {
+    results.emplace_back(getInputs()[0]);
```

Comment:
I would use `push_back` here, no need for `emplace_back`.

Reply:
Thanks. updated as suggested.